standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/JdoPropertyStore.java

```diff
@@ -0,0 +1,154 @@
+    try {
+      return objectStore.getProperties(mapKey, getPropertyMapFunction(null, getSchema));
+    } catch (MetaException | JexlException e) {
+      throw new PropertyException(e);
+    }
+  }
+
+  @Override
+  public Map<String, PropertyMap> selectProperties(final String keyPrefix, Predicate<String> keyFilter, Function<String, PropertySchema> getSchema) {
+    try {
+      return objectStore.selectProperties(keyPrefix, getPropertyMapFunction(keyFilter, getSchema));
+    } catch (MetaException | JexlException e) {
+      throw new PropertyException(e);
+    }
+  }
+
+  @Override
+  public UUID fetchDigest(String mapKey) {
+    try {
+      return objectStore.getProperties(mapKey, (mm) -> UUID.fromString(mm.getPropertyValue()));
+    } catch (MetaException | JexlException e) {
+      throw new PropertyException(e);
+    }
+  }
+
+  @Override
+  public Map<String, UUID> selectDigest(String keyPrefix, Predicate<String> keyFilter) {
+    try {
+      return objectStore.selectProperties(keyPrefix, (mm) -> {
+        if (keyFilter == null || keyFilter.test(mm.getPropertykey())) {
+          return UUID.fromString(mm.getPropertyValue());
+        }
+        return null;
+      });
+    } catch (MetaException | JexlException e) {
+      throw new PropertyException(e);
+    }
+  }
+
+  @Override
+  public void saveProperties(Iterator<Map.Entry<String, PropertyMap>> save) {
+    // will run the super method in a transaction
+    try {
+      objectStore.runInTransaction(()-> super.saveProperties(save));
+    } catch (MetaException e) {
+      throw new PropertyException(e);
+    }
+  }
+
+  @Override
+  protected void saveProperties(String mapKey, PropertyMap map) {
+    try {
+      if (map.isDropped()) {
+        objectStore.dropProperties(mapKey);
+      } else {
+        objectStore.putProperties(mapKey, map.getDigest().toString(), null, serialize(map));
+      }
+    } catch (MetaException e) {
+      throw new PropertyException(e);
+    }
+  }
+
+  @Override public boolean dropProperties(String mapKey) {
+   try {
+     return objectStore.dropProperties(mapKey);
+   } catch (MetaException e) {
+     throw new PropertyException(e);
+   }
+  }
+
+  @Override public boolean renameProperties(String mapKey, String newKey) {
+    try {
+      return objectStore.renameProperties(mapKey, newKey);
+    } catch (MetaException e) {
+      throw new PropertyException(e);
+    }
+  }
+  /**
+   * Creates a function that transforms an MMetastoreDBProperties into a PropertyMap.
+   * @param keyFilter a map key filtering predicate that will make the function return null if test fails
+   * @param getSchema the function that solves a schema from a key
+   * @return a function
+   */
+  Function<MMetastoreDBProperties, PropertyMap> getPropertyMapFunction(final Predicate<String> keyFilter, final Function<String, PropertySchema> getSchema) {
+    return (mm) -> {
+      final String key = mm.getPropertykey();
```

Comment:
Why the final keyword is necessary here in this method?

Reply:
Not necessary, just stating it won't change after init. In a method returning a lambda, I tried to be extra-clear.

Reply:
ack